### PR TITLE
*: add NodeTrimThreshold to QueryMerge requests

### PR DESCRIPTION
**Pull request by asubiotto**

### CODE REVIEW

**Comment by asubiotto** [comment]
I'm not sure why the NodeTrimThreshold is a pointer

**Review by brancz — APPROVED** [review]

**Review by metalmatze — COMMENTED** [review]
Good one!

File: querier.go [hunk 1]
```diff
@@ -18,6 +18,10 @@ import (
 	"google.golang.org/protobuf/types/known/timestamppb"
 )
 
+// Trim anything that can't be displayed by an 8K display (7680 horizontal
+// pixels). This is a var because the proto request requires an address.
+var nodeTrimThreshold = float32(1) / 7680
```
[the remark this answers is not in the document]
Comment (brancz) [reply querier.go:23]: I think this is a good value to start with. It's certainly quite pessimistic, but less than a 4k display width would be not testing things that happen a lot.

[the remark this answers is not in the document]
Comment (asubiotto) [reply querier.go:23]: Yes but I think we want this to be representative of what a user that does look at the data experiences.

**Comment by brancz** [comment]
> I'm not sure why the NodeTrimThreshold is a pointer

Optional field in proto?

File: querier.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (metalmatze) [reply querier.go:23]: Ok, fair enough. :+1:

**Review by metalmatze — APPROVED** [review]

**Comment by asubiotto** [comment]
> > I'm not sure why the NodeTrimThreshold is a pointer
> 
> Optional field in proto?

Sure, but why is it optional? Wouldn't a value of 0 be the same as not specifying it? This is more of a nit

**Comment by metalmatze** [comment]
It could default to `0` as being optional. It's just that we use the optional field feature of protobuf here. 
Given that this is unrelated to the change in parca-load itself, I'd say we go ahead and merge this. 

If that's something worth fixing, we should do that in [Parca](https://github.com/parca-dev/parca/blob/adbbf003396177b295f5c5ad4c2fe228ee9ed3ae/proto/parca/query/v1alpha1/query.proto#L242).

**Comment by asubiotto** [comment]
Does the step change look good as well? Will merge if so

File: querier.go [hunk 2]
```diff
@@ -233,6 +240,7 @@ func (q *Querier) queryRange(ctx context.Context) {
 			Query: profileType,
 			Start: timestamppb.New(rangeStart),
 			End:   timestamppb.New(rangeEnd),
+			Step:  durationpb.New(time.Duration(tr.Nanoseconds() / numHorizontalPixelsOn8KDisplay)),
```
Comment (metalmatze) [inline querier.go:243]: The step parameter for QueryRange only depends on the start and end time in the query.
https://github.com/parca-dev/parca/blob/main/ui/packages/shared/utilities/src/time.ts#L129-L135

I think we need to adjust to use the `(end - start) / stepCount`. We can make our step count the `numHorizontalPixelsOn8KDisplay`.

Comment (asubiotto) [reply querier.go:243]: Thanks for the pointer! In this case `tr = end-start` so I think this does what's expected.

**Pull request merged**